specs/schemas/schema-schema.ipldsch

```diff
@@ -423,17 +423,23 @@ type UnionRepresentation_Inline struct {
 	discriminantTable {String:TypeName}
 }
 
-## UnionRepresentation_BytePrefix describes a union representation for unions
+## UnionRepresentation_BytesPrefix describes a union representation for unions
 ## whose member types are all bytes. It is encoded to a byte array whose
-## first byte is the discriminator and subsequent bytes form the discriminated
+## first bytes are the discriminator and subsequent bytes form the discriminated
 ## type.
 ##
-## byteprefix is an invalid representation for any union that contains a type
+## Discriminators are represented as hexadecimal strings. There is currently
+## no limitation on their length, other than needing to be at least one byte.
+## Nor is there a requirement that they all be of the same length, although
+## they must all represent unique prefixes.
+##
+## Only valid, upper-case, hexadecimal strings representing at least one byte
+## are allowed.
+##
+## bytesprefix is an invalid representation for any union that contains a type
 ## that does not have a bytes representation.
 ##
-type UnionRepresentation_BytePrefix struct {
-	discriminantTable {TypeName:Int}
-}
+type UnionRepresentation_BytesPrefix {String:TypeName}
```

Comment:
Just highlighting for the sake of clarity: this does defacto give up any possibility to add more features to UnionRepresentation_BytePrefix in the future.

But this is probably fine, because A) I don't have even speculative ideas today about what those "more features" could be, and B) worst case scenario, we can still add new members to UnionRepresentation union in the future, so we've got another flex point near to hand if we really need it.